heartbeat/monitors/wrappers/util.go

```diff
@@ -58,7 +58,15 @@ func URLFields(u *url.URL) common.MapStr {
 	}
 
 	if u.Port() != "" {
-		fields["port"], _ = strconv.ParseUint(u.Port(), 10, 16)
+		// Returns a uint64 believe it or not.
+		val, _ := strconv.ParseUint(u.Port(), 10, 16)
+		fields["port"] = uint16(val)
+	} else {
+		if u.Scheme == "http" {
+			fields["port"] = uint16(80)
+		} else if u.Scheme == "https" {
+			fields["port"] = uint16(443)
```

Comment:
Should we have an ending "else" to catch edge cases we were not thinking of?

Reply:
Good question. I believe the answer is no. 

ICMP does not have a port
TCP has no notion of a default port, and our configs require one to be set anyway.